projects/js-packages/publicize-components/src/social-store/resolvers.js

```diff
@@ -72,6 +73,12 @@ export function getPostShareStatus( _postId ) {
 	return async ( { dispatch, registry } ) => {
 		// Default to the current post ID if none is provided.
 		const postId = _postId || registry.select( editorStore ).getCurrentPostId();
+		const featureFlags = registry.select( socialStore ).featureFlags();
+
+		if ( ! featureFlags.useShareStatus ) {
+			dispatch( fetchPostShareStatus( postId, false ) );
```

Comment:
I don't think you need to dispatch this action here. A simple return should work fine.